docs/NewFlux_Comentary.md

```diff
@@ -0,0 +1,252 @@
+
+### Aggregates and Selectors
+
+With the change to a single table model, aggregates and selectors are now expressed differently.
+There is a new function `aggregate` that also comes with its own syntax. More on the syntax in a bit.
+
+The `aggregate` function has a `by` parameter which specifies the dimensions on which you would like to perform an aggregate function.
+Similarly the `select` function has a `by` parameter. 
+
+
+These functions now make it explicit how you want to perform an operation and that the grouping of that operation is scoped to just that operation.
+No longer is there a persistent state on the table that indicates how the table (or stream of tables) are grouped.
+
+Similarly the `aggregate` and `select` functions will have parameters for `windowing`, that enable you to define how you would like to group the data in time.
+
+The `group` and `window` functions are not part of this new Flux vision.
+
+### Syntax Sugar for Aggregates
+
+Aggregates also come with their own new syntax. The intent of this syntax is to make it easy to describe which aggregate function to apply to which columns.
+This syntax allows for that definition to be explicit and terse.
+
+The syntax is:
+
+```
+ |> aggregate({min(degrees_bottom), max(degrees_bottom), mean(degrees_bottom)}, by:["location"])
+```
+
+The semantics of this syntax are to apply three different aggregate functions to the same column named `degrees_bottom`.
+The output table will then have three columns named `min_degrees_bottom`, `max_degrees_bottom`, and `mean_degrees_bottom`.
+
+
+>NOTE: Selectors do not need their own syntax as it is easy to specify selection in the existing syntax.
+
+### Column Indexing
+
+A consequence of data being represented as a single table the `tableFind` function is not needed. As such  a user can call `getColumn` directly on a table.
+We can go one step further and use the `.` syntax to mean indexing a table for its column.
+
+For example:
+
+```
+data = @foo.bar{location, state == "CA"}
+
+data.location // a list of locations in the state of CA
+```
+
+
+## The How
+
+Now that we have an overview of the new features syntax lets discuss how I expect they can be implemented.
+Again this is just to seed the conversation and is by no means a prescription on how it must be done.
+
+
+### Single Table vs Stream of Tables
+
+The current Flux internals use the interpreter to create DAG of promises to data to construct a query spec.
+That spec is then planned etc and passed down to the execution engine.
+
+So long as we can continue to build a query spec from this new Flux API we can continue to execute the queries.
+My rough plan is that a set of new Flux functions will be added to a `table` package.
+The `table` package will be very similar to the current `universe` package, it will contain all the basic functions and the new functions as well like `aggregate`.
```

Comment:
```suggestion
The `table` package will be very similar to the current `universe` package. It will contain all basic functions and new functions as well like `aggregate`.
```